py-polars/polars/functions/lit.py

```diff
@@ -101,7 +101,8 @@ def lit(
             return e
 
     elif isinstance(value, timedelta):
-        time_unit = "us" if dtype is None else getattr(dtype, "time_unit", "us")
+        if dtype is None or (time_unit := getattr(dtype, "time_unit", "us")) is None:
```

Comment:
@MarcoGorelli note the `time_unit` attribute *does* exist for regular `pl.Duration` even if the time unit isn't specified, so in that case the default `"us"` is *not* returned, the `None` value is. So we have to check for that too.